### PR TITLE
community: removed unnecessary dependecy use from NotebookLoader

### PR DESCRIPTION
**Pull request by lachiewalker**
**Description:** Removes unnecessary pandas use from the NotebookLoader document loader.
- In no way changes the output from this classes `.load()` method.
- Is compatible with existing tests.
- Delivers a noticeable speedup over the previous method.

### CODE REVIEW

File: libs/community/langchain_community/document_loaders/notebook.py [hunk 1]
```diff
@@ -104,29 +108,29 @@ def load(
...
-            raise ImportError(
-                "pandas is needed for Notebook Loader, "
-                "please install with `pip install pandas`"
-            )
         p = Path(self.file_path)
 
         with open(p, encoding="utf8") as f:
             d = json.load(f)
 
-        data = pd.json_normalize(d["cells"])
```
Comment (baskaryan) [inline libs/community/langchain_community/document_loaders/notebook.py:119]: what does pd.json_normalize do

Comment (lachiewalker) [reply libs/community/langchain_community/document_loaders/notebook.py:119]: From the [documentation](https://pandas.pydata.org/docs/reference/api/pandas.json_normalize.html): Normalizes semi-structured JSON data into a flat table.

Combined with `filtered_data = data[["cell_type", "source", "outputs"]]` this statement makes a table of [ cell_type, source, output ] information.

This behaviour can be simply replicated by a list of dictionaries with keys `cell_type`, `source`, `output`, without relying on a dependency.

Comment (baskaryan) [reply libs/community/langchain_community/document_loaders/notebook.py:119]: makes sense, could we add a simple unit test?

Comment (paulonasc) [reply libs/community/langchain_community/document_loaders/notebook.py:119]: @lachiewalker 👋 new to contributing here, I was wondering if I can help write the unit test?

Comment (lachiewalker) [reply libs/community/langchain_community/document_loaders/notebook.py:119]: @paulonasc yeah, of course 🙂

Comment (paulonasc) [reply libs/community/langchain_community/document_loaders/notebook.py:119]: here's the commit for the unit tests btw! https://github.com/langchain-ai/langchain/pull/17721

Is it okay that it's in a separate PR?

**Pull request closed without merging**